src/manifolds/MultinomialSymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,52 @@
+@doc raw"""
+    MultinomialSymmetricPositiveDefinite <: AbstractMultinomialDoublyStochastic
+
+The symmetric positive definite multinomial matrices manifold consists of all
+symmetric ``n×n`` matrices with positive eigenvalues, and
+positive entries such that each column sums to one, i.e.
+
+````math
+\begin{aligned}
+\mathcal{SP}^+(n) \coloneqq \bigl\{
+    p ∈ ℝ^{n×n}\ \big|\ &p_{i,j} > 0 \text{ for all } i=1,…,n, j=1,…,m,\\
+& p^\mathrm{T} = p,\\
+& p\mathbf{1}_n = \mathbf{1}_n\\
+a^\mathrm{T}pa > 0 \text{ for all } a ∈ ℝ^{n}\backslash\{\mathbf{0}_n\}
+\bigr\},
+\end{aligned}
+````
+
+where ``\mathbf{1}_n`` and ``\mathbr{0}_n`` are the vectors of length ``n``
+containing ones and zeros, respectively. More details for this manifold can b found in
+[DouikHassibi:2019](@cite).
+
+# Constructor
+
+    MultinomialSymmetricPositiveDefinite(n)
+
+Generate the manifold of matrices ``\mathbb R^{n×n}`` that are symmetric, positive definite, and doubly stochastic.
+"""
+struct MultinomialSymmetricPositiveDefinite{T} <: AbstractMultinomialDoublyStochastic
```

Comment:
Could we have a `rand` for these too?

Reply:
I would have to check, whether I see them somewhere, for now I just saw that manifold in the paper (where we have the other three from).
If I see a good wax to construct there, sure.

Reply:
Probably generating a random totally positive matrix: https://math.stackexchange.com/questions/2773460/how-to-generate-a-totally-positive-matrix-randomly-using-software-like-maple and then projecting to `AbstractMultinomialDoublyStochastic` would work.

Reply:
That would at least not directly yield an error, but would the result have positive eigenvalues? I am not so sure, but I can check.

Reply:
I fiddled a bit with the random for MSPD, I fear I am not clever enough for this. Not only does it involve several of our manifolds (symmetric, SPDs, multinomial, Simplex, and all might throw errors somewhere along the way), I can just not get (a) random to work with the positive matrix for points nor the one for the vectors, since I am not sure how the projection looks like.

So in theory – yes. In practice – not.

Reply:
So if you can do that, feel free to implement `project` for vectors and rand including your idea for the randomness – to which in theory I agree – just that in practice I can not get it to work (but I also did a whole day of writing tests, so my brain is not much more than a pudding by now).

Reply:
I've written `rand` based on that post, seems to work:
```julia
function rand(M::MultinomialSymmetricPositiveDefinite)
    n = get_parameter(M.size)[1]
    L = sort(exp.(randn(n)))
    V = reduce(hcat, map(xi -> [xi^k for k in 0:(n-1)], L))'
    Vlu = lu(V, LinearAlgebra.RowNonZero())
    dm = Diagonal(Vlu.U)
    uutd = dm \ Vlu.U
    random_totally_positive = uutd * dm * Vlu.L
    MMDS = MultinomialDoubleStochastic(n)
    ds = project(MMDS, random_totally_positive)
    sym = (ds .+ ds')./2
    return sym
end
```

Reply:
I will add some documentation later because it's quite involved.

Reply:
That looks nearly as involved as the Hessian I did. 
And we should do that as a `rand!(rng, M, p; kwargs...)` (maybe we see how to do tangent vectors as well :) )

What – at first glance – I did not have, was the exp

Reply:
`exp` is not strictly necessary, we only need `L` to be positive and sorted. About 2/3 of the work was figuring out how to do the exact variant of LDU decomposition it needs.

Reply:
I've added it to the PR. I haven't checked tangent vector generation yet but that seems to be less important.

Reply:
If I see that correctly, Symmetric Multonomial should do the job.

Reply:
Sure I can check. I wanted to finish the other PR first anyways (the other one has so many strings changed in the docs, so lets better merge it here, i.e. finish that first), so this one is not that urgent.